components/sidebar/sidebar_service_unittest.cc

```diff
@@ -729,4 +877,175 @@ TEST_F(SidebarServiceTestWithPlaylist, GetDefaultPanelItem) {
   EXPECT_FALSE(service_->GetDefaultPanelItem());
 }
 
+class SidebarServiceOrderingTest : public testing::Test {
+ public:
+  SidebarServiceOrderingTest() = default;
+  SidebarServiceOrderingTest(const SidebarServiceOrderingTest&) = delete;
+  SidebarServiceOrderingTest& operator=(const SidebarServiceOrderingTest&) =
+      delete;
+  SidebarServiceOrderingTest(const SidebarServiceOrderingTest&&) = delete;
+  SidebarServiceOrderingTest& operator=(const SidebarServiceOrderingTest&&) =
+      delete;
+
+  ~SidebarServiceOrderingTest() override = default;
+
+  PrefService* GetPrefs() { return profile_->GetPrefs(); }
+
+  SidebarService* GetSidebarService() { return sidebar_service_; }
+
+  void SetUp() override {
+    testing::Test::SetUp();
+
+#if BUILDFLAG(ENABLE_AI_CHAT)
+    scoped_feature_list_.InitAndEnableFeature(ai_chat::features::kAIChat);
+#endif  // BUILDFLAG(ENABLE_AI_CHAT)
+
+    auto registry = base::MakeRefCounted<user_prefs::PrefRegistrySyncable>();
+    sidebar::SidebarServiceFactory::GetInstance();
+    RegisterUserProfilePrefs(registry.get());
+    registry->RegisterDictionaryPref("brave");
+    registry->RegisterDictionaryPref("brave.sidebar");
+
+    sync_preferences::PrefServiceMockFactory factory;
+    auto pref_service = factory.CreateSyncable(registry.get());
+    auto* pref_service_ptr = pref_service.get();
+
+    auto builder = TestingProfile::Builder();
+    builder.SetPrefService(std::move(pref_service));
+    profile_ = builder.Build();
+
+    ASSERT_EQ(pref_service_ptr, profile_->GetPrefs());
```

Comment:
What's the reason for doing this instead of having a simpler `TestingPrefServiceSimple` and calling SidebarService::RegisterProfilePrefs`? Perhaps add a comment here as to why it's this way.

Reply:
I think you could subclass `SidebarServiceTest` and add additional code for testing ai chat
as `SidebarServiceTest` already has similar `InitService()`.

Reply:
done